### PR TITLE
lib: Consolidate tests into one meta job

### PR DESCRIPTION
**Pull request by Ericson2314**
###### Motivation for this change

@edolstra is this what you want?

### CODE REVIEW

**Comment by matthewbauer** [comment]
Make sure you rename it in make-tarball.nix too:

https://github.com/NixOS/nixpkgs/blob/7a586794d4378d7a6432192700a4706083d58388/pkgs/top-level/make-tarball.nix#L60

**Comment by Ericson2314** [comment]
Oh heh I didn't even see that. Maybe I'll just make the lib tests a sub job instead.

**Comment by edolstra** [comment]
Looks good to me, thanks!

File: lib/tests/release.nix [hunk 1]
```diff
@@ -1,40 +1,32 @@
-{ nixpkgs ? { outPath = (import ../.).cleanSource ../..; revCount = 1234; shortRev = "abcdef"; }
-, # The platforms for which we build Nixpkgs.
-  supportedSystems ? [ builtins.currentSystem ]
-, # Strip most of attributes when evaluating to spare memory usage
-  scrubJobs ? true
-}:
+{ pkgs ? import ((import ../../lib).cleanSource ../..) {} }:
```
Comment (nbp) [inline lib/tests/release.nix]: You will have to update the list of attribute given when this expression is imported in `pkgs/top-level/release.nix` to make this evaluate correctly.

**Comment by Ericson2314** [comment]
- Mac tests not started, but other two passed, and no platform-specific code here
- Fixed issues, thanks @matthewbauer and @nbp

**Pull request merged**